jupyter_server/terminal/api_handlers.py

```diff
@@ -21,7 +21,11 @@ def get(self):
     @web.authenticated
     def post(self):
         """POST /terminals creates a new terminal and redirects to it"""
-        name, _ = self.terminal_manager.new_named_terminal()
+        data = self.get_json_body()
+        if data and data.get('cwd'):
+            cwd = data['cwd']
+
+        name, _ = self.terminal_manager.new_named_terminal(cwd=cwd)
```

Comment:
Won't this fail if no JSON body is given since `cwd` won't be defined?

Reply:
I changed this and added tests. Please take a look

Reply:
Looks good :+1: